manager/manager.go

```diff
@@ -727,3 +526,232 @@ func (m *Manager) rotateRootCAKEK(ctx context.Context, clusterID string) error {
 	})
 
 }
+
+// handleLeadershipEvent read out and discard all of the messages when the manager is stopped,
+// otherwise it handles the is leader event or is follower event.
+func (m *Manager) handleLeadershipEvents(ctx context.Context, leadershipCh chan events.Event) {
+	for leadershipEvent := range leadershipCh {
+		// read out and discard all of the messages when we've stopped
+		// don't acquire the mutex yet. if stopped is closed, we don't need
+		// this stops this loop from starving Run()'s attempt to Lock
+		select {
+		case <-m.stopped:
+			continue
+		default:
+			// do nothing, we're not stopped
+		}
+		// we're not stopping so NOW acquire the mutex
+		m.mu.Lock()
+		newState := leadershipEvent.(raft.LeadershipState)
+
+		if newState == raft.IsLeader {
+			m.becomeLeader(ctx)
+		} else if newState == raft.IsFollower {
+			m.becomeFollower()
+		}
+		m.mu.Unlock()
+	}
+}
+
+// serveListener serves a listener for local and non local connections.
+func (m *Manager) serveListener(ctx context.Context, errServe chan error, proto string, lis net.Listener) {
+	ctx = log.WithLogger(ctx, log.G(ctx).WithFields(
+		logrus.Fields{
+			"proto": lis.Addr().Network(),
+			"addr":  lis.Addr().String()}))
+	if proto == "unix" {
+		log.G(ctx).Info("Listening for local connections")
+		// we need to disallow double closes because UnixListener.Close
+		// can delete unix-socket file of newer listener. grpc calls
+		// Close twice indeed: in Serve and in Stop.
+		errServe <- m.localserver.Serve(&closeOnceListener{Listener: lis})
+	} else {
+		log.G(ctx).Info("Listening for connections")
+		errServe <- m.server.Serve(lis)
+	}
+}
+
+// becomeLeader  promote itself to be a leader,
```

Comment:
I think it's okay to delete it. The specifics should be explained by the code itself.